src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageHandler.java

```diff
@@ -52,8 +62,16 @@
  * the next handler (i.e. throws exception instead), if it cannot process
  * the request itself.
  */
+@Slf4j
 class ClientRestMessageHandler extends AbstractClientProxyHandler {
 
+    private static final String TEXT_XML = "text/xml";
+    private static final String APPLICATION_XML = "application/xml";
+    private static final String TEXT_ANY = "text/*";
+    private static final String APPLICATION_JSON = "application/json";
+    private String requestAcceptType;
```

Comment:
requestAcceptType can be overwritten by a simultaneous request. The ClientRestMessageHandler instance is shared by all requests; it can not have per-request state.

Reply:
Fixed